hudi-utilities/src/main/java/org/apache/hudi/utilities/checkpointing/InitialCheckPointProvider.java

```diff
@@ -20,12 +20,23 @@
 
 import org.apache.hudi.exception.HoodieException;
 
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
 /**
  * Provide the initial checkpoint for delta streamer.
  */
-public interface InitialCheckPointProvider {
+public abstract class InitialCheckPointProvider {
+  protected final Path path;
+  protected final FileSystem fs;
+
+  public InitialCheckPointProvider(final Path basePath, final FileSystem fileSystem) {
```

Comment:
it may be worth turning this into  `abstract void init(Path, FileSystem)` method? that way all subclasses are forced to implement the right one.. 

Also can we pass the `props` or the master list of properties (like we do for key generators) into this abstraction and let the CheckpointProvider use an explicit property like `hoodie.deltastreamer.checkpointprovider.kafka.connect.path` to derive the bootstrap path

Reply:
Adding `props` is a good idea. It will add more flexibility for other providers.